doc/Learn/Mvux/Advanced/Messaging.md

```diff
@@ -302,6 +302,81 @@ The `SelectedPersonPhone` state will only be refreshed if it meets the predicate
 
 This overload is the same as the previous one, except it watches a single-item state rather than a `ListState`, as in the last example.
 
+#### Fluent API Observe overloads
+
+These extension methods are available for both `IState<TEntity>` and `IListState<TEntity>` objects.
+
+- `IState<TEntity> Observe<TEntity, TKey>(this IState<TEntity> state, IMessenger messenger, Func<TEntity, TKey> keySelector)`
+
+This overload is a fluent API extension method that can be used to observe entity-change messages from the messenger for a specific entity type and apply them to the state. It returns the state itself, so it can be chained with other methods.
+
+```csharp
+public partial record MyModel
+{
+    protected IUserService UserService { get; }
+
+    public MyModel(IUserService userService, IMessenger messenger)
+    {
+        UserService = userService;
+
+        CurrentUser
+            .Observe(messenger, user => user.Id)
+            .Observe(messenger, user => user.Name);
+    }
+
+    public IState<User> CurrentUser => State.Async(this, UserService.GetCurrentUser);
+}
+```
+
+or in a more Fluent API way:
+
+```csharp
+public partial record MyModel(IUserService UserService, IMessenger Messenger)
+{
+    public IState<User> CurrentUser => State.Async(this, UserService.GetCurrentUser)
+                                            .Observe(Messenger, user => user.Id)
+                                            .Observe(Messenger, user => user.Name);
```

Comment:
why would you align this way?
```suggestion
    public IState<User> CurrentUser => State.Async(this, UserService.GetCurrentUser)
        .Observe(Messenger, user => user.Id)
        .Observe(Messenger, user => user.Name);
```

Reply:
That's an old habit. This is how I usually align in my code.